src/Blockcore.Indexer.Core/Storage/Mongo/MongoData.cs

```diff
@@ -922,87 +886,34 @@ public QueryResult<QueryTransaction> GetMemoryTransactions(int offset, int limit
-            {
-               Address = utxo.Address,
-               Outpoint = utxo.Outpoint,
-               Value = utxo.Value,
-               BlockIndex = utxo.BlockIndex,
-               CoinBase = utxo.CoinBase,
-               CoinStake = utxo.CoinStake,
-               ScriptHex = utxo.ScriptHex
-            });
-
-         return new QueryResult<UnspentOutputsView>
-         {
-            Items = utxos,
-            Total = total,
-            Offset = offset,
-            Limit = limit
-         };
-
-      }
-
-      public async Task<QueryResult<UnspentOutputsView>> GetUnspentTransactionsByAddressAsync_Old(string address, long confirmations, int offset, int limit)
-      {
-         var totalTask = Task.Run(() => OutputTable.Aggregate()
+         var totalTask = Task.Run(() => UnspentOutputTable.Aggregate()
             .Match(_ => _.Address.Equals(address))
             .Match(_ => _.BlockIndex <= globalState.StoreTip.BlockIndex - confirmations)
-            .Lookup(InputTable.CollectionNamespace.CollectionName,
-               new StringFieldDefinition<OutputTable>(nameof(Outpoint)),
-               new StringFieldDefinition<BsonDocument>(nameof(Outpoint)),
-               new StringFieldDefinition<BsonDocument>("Inputs"))
-            .Match(_ => _["Inputs"] == new BsonArray())
             .Count()
             .Single());
 
-         var selectedTask = Task.Run(() => OutputTable.Aggregate()
+         var outpointsToFetchTask = Task.Run(() => UnspentOutputTable.Aggregate()
             .Match(_ => _.Address.Equals(address))
             .Match(_ => _.BlockIndex <= globalState.StoreTip.BlockIndex - confirmations)
-            .Sort(new BsonDocumentSortDefinition<OutputTable>(new BsonDocument("BlockIndex",-1)))
-            .Lookup(InputTable.CollectionNamespace.CollectionName,
-               new StringFieldDefinition<OutputTable>(nameof(Outpoint)),
-               new StringFieldDefinition<BsonDocument>(nameof(Outpoint)),
-               new StringFieldDefinition<BsonDocument>("Inputs"))
-            .Match(_ => _["Inputs"] == new BsonArray())
+            .Sort(new BsonDocumentSortDefinition<UnspentOutputTable>(new BsonDocument("BlockIndex",-1))) //TODO David, will need to chane -1 when the index is changed to descending
             .Skip(offset * limit)
             .Limit(limit)
             .ToList()
-            .Select(_ => new UnspentOutputsView
-            {
-               Address = _["Address"].AsString,
-               Outpoint = new Outpoint
-               {
-                  OutputIndex = _["Outpoint"]["OutputIndex"].AsInt32,
-                  TransactionId = _["Outpoint"]["TransactionId"].AsString,
-               }  ,
-               Value = _["Value"].AsInt64,
-               BlockIndex = _["BlockIndex"].AsInt64,
-               CoinBase = _["CoinBase"].AsBoolean,
-               CoinStake = _["CoinStake"].AsBoolean,
-               ScriptHex = _["ScriptHex"].AsString
```

Comment:
What about `ScriptHex` we are suppose to look it up no?
This can be done in another PR though